openedx/features/learner_profile/views/learner_achievements.py

```diff
@@ -0,0 +1,42 @@
+"""
+Views to render a learner's achievements.
+"""
+
+from courseware.courses import get_course_overview_with_access
+from django.template.loader import render_to_string
+from lms.djangoapps.certificates import api as certificate_api
+from openedx.core.djangoapps.plugin_api.views import EdxFragmentView
+from web_fragments.fragment import Fragment
+
+
+class LearnerAchievementsFragmentView(EdxFragmentView):
+    """
+    A fragment to render a learner's achievements.
+    """
+    def render_to_fragment(self, request, username=None, own_profile=False, **kwargs):
+        """
+        Renders the current learner's achievements.
+        """
+        course_certificates = self._get_ordered_certificates_for_user(request, username)
+        context = {
+            'course_certificates': course_certificates,
+            'own_profile': own_profile,
+            'disable_courseware_js': True,
+        }
+        if course_certificates or own_profile:
+            html = render_to_string('learner_profile/learner-achievements-fragment.html', context)
+            return Fragment(html)
+        else:
+            return None
+
+    def _get_ordered_certificates_for_user(self, request, username):
+        """
+        Returns a user's certificates sorted by course name.
+        """
+        course_certificates = certificate_api.get_certificates_for_user(username)
```

Comment:
@andy-armstrong I think we still need to filter this list by site as @robrap mentioned previously.

Reply:
Thanks @douglashall. @robrap has a subtask to make this work.

Reply:
As discussed, we'll follow up with this immediately in the new story.

Reply:
Here's the ticket: https://openedx.atlassian.net/browse/LEARNER-2449